src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -617,9 +615,16 @@ private SyntaxNode FindStatement(SyntaxNode declarationBody, int position, out i
                         continue;
                     }
 
+                    // Finds a matching syntax node in the new source.
+                    // In case the node got deleted the newMember may be missing.
+                    // For those the active span should remain empty.
                     SyntaxNode newMember;
                     bool hasPartner = topMatch.TryGetNewNode(oldMember, out newMember);
-                    Debug.Assert(hasPartner);
+                    Debug.Assert(hasPartner || trackedSpan.IsEmpty);
```

Comment:
The change is necessary. The span of a deleted top level syntax node is empty. Therefore `UpdateUneditedSpans` expected this node to be unedited and have a matching syntax node in the new tree. This caused a `NullPointerException` and was the reason for VS crashing.